packages/fetchai/connections/http/connection.py

```diff
@@ -0,0 +1,359 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2020 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""HTTP connection, channel, and server"""
+
+import asyncio
+import logging
+import threading
+import time
+import yaml
+import json
+from asyncio import CancelledError
+from threading import Thread
+from typing import Any, Dict, List, Optional, Set, cast
+from http.server import HTTPServer, BaseHTTPRequestHandler
+
+from aea.configurations.base import ConnectionConfig, PublicId
+from aea.connections.base import Connection
+from aea.mail.base import Address, Envelope, EnvelopeContext, URI
+
+
+logger = logging.getLogger(__name__)
+
+
+class HTTPHandler(BaseHTTPRequestHandler):
+
+    def do_HEAD(s):
+        # Placeholder functions
+        s.send_response(200)
+        s.send_header("Content-type", "text/html")
+        s.end_headers()
+
+    def do_GET(s):
+        """Respond to a GET request."""
+        # Placeholder functions
```

Comment:
hear is where you'd create the envelope, send it in to the agent, wait for the response envelope and then serve the response